src/views/Details/Details.js

```diff
@@ -121,18 +127,27 @@ export default {
             'payments',
             'accounts',
             'isPaymentLoading',
+            'currencies'
         ]),
         account() {
             return this.accounts.find(a => a.id === this.id);
         },
         accountPayments() {
-            return this.payments(this.id);
+            return this.payments(this.id).filter(payment => {
+                if(this.selectedCurrency == 'All' || payment.asset_code == this.selectedCurrency){
```

Comment:
When selected currency is null this could be more easily checked e.g. `if(!this.selectedCurrency) {return this.payments(this.id)};  '